In [ ]:
###################

Student Name: Jing Zhu
Project: HAP725_W2Q2_HospitalCompareData
    
##################

In [7]:
import pandas as pd

# 1. Data Description

In this case we mainly calculate the standard deviation for the rate across all hospitals and report the average rate the standard deviation of the rate from the the Medicare Compare site.

In [9]:
jzdata = pd.read_csv('Complications_Hospital.csv')

In [10]:
jzdata.head()

Provider ID                     Hospital Name                 Address  \
0        10001  SOUTHEAST ALABAMA MEDICAL CENTER  1108 ROSS CLARK CIRCLE   
1        10001  SOUTHEAST ALABAMA MEDICAL CENTER  1108 ROSS CLARK CIRCLE   
2        10001  SOUTHEAST ALABAMA MEDICAL CENTER  1108 ROSS CLARK CIRCLE   
3        10001  SOUTHEAST ALABAMA MEDICAL CENTER  1108 ROSS CLARK CIRCLE   
4        10001  SOUTHEAST ALABAMA MEDICAL CENTER  1108 ROSS CLARK CIRCLE   

     City State  ZIP Code County Name  Phone Number  \
0  DOTHAN    AL     36301     HOUSTON    3347938701   
1  DOTHAN    AL     36301     HOUSTON    3347938701   
2  DOTHAN    AL     36301     HOUSTON    3347938701   
3  DOTHAN    AL     36301     HOUSTON    3347938701   
4  DOTHAN    AL     36301     HOUSTON    3347938701   

                                        Measure Name  \
0  Rate of complications for hip/knee replacement...   
1                  Serious blood clots after surgery   
2               Blood stream infection after surgery   
3  A wound that splits open  after surgery on the...   
4   Accidental cuts and tears from medical treatment   

                  Measure ID                 Compared to National Denominator  \
0              COMP_HIP_KNEE  No Different than the National Rate         331   
1  PSI_12_POSTOP_PULMEMB_DVT  No Different than the National Rate        4549   
2         PSI_13_POST_SEPSIS  No Different than the National Rate         578   
3        PSI_14_POSTOP_DEHIS  No Different than the National Rate         445   
4             PSI_15_ACC_LAC  No Different than the National Rate       16429   

  Score Lower Estimate Higher Estimate Footnote Measure Start Date  \
0   3.8            2.6             5.5      NaN         04/01/2012   
1  3.32           1.25            5.38      NaN         07/01/2013   
2  5.72              0           11.48      NaN         07/01/2013   
3  2.72           0.65            4.79      NaN         07/01/2013   
4  1.35           0.76            1.95      NaN         07/01/2013   

  Measure End Date  
0       03/31/2015  
1       06/30/2015  
2       06/30/2015  
3       06/30/2015  
4       06/30/2015

In [11]:
print('Number of instances = %d'% (jzdata.shape[0])) 
print('Number of attributes = %d'% (jzdata.shape[1]))

Number of instances = 52877
Number of attributes = 18


## The data contains 18 columns and 52877 rows.

# 2. Filter Data

In [12]:
filtered = jzdata[jzdata['Measure ID'] == 'COMP_HIP_KNEE']

In [13]:
filtered.dtypes

Provider ID              int64
Hospital Name           object
Address                 object
City                    object
State                   object
ZIP Code                 int64
County Name             object
Phone Number             int64
Measure Name            object
Measure ID              object
Compared to National    object
Denominator             object
Score                   object
Lower Estimate          object
Higher Estimate         object
Footnote                object
Measure Start Date      object
Measure End Date        object
dtype: object

In [14]:
score = pd.to_numeric(filtered['Score'], errors='coerce')

In [15]:
score.head(5)

0     3.8
11    3.0
22    3.8
33    NaN
44    NaN
Name: Score, dtype: float64

# 3. Calculate Average Rate

In [16]:
score.mean()

3.0415560471976497

In [17]:
score.std()

0.5596498066792593